backend/ecs_tasks/delete_files/cse.py

```diff
@@ -0,0 +1,138 @@
+import base64
+import json
+import logging
+import os
+from io import BytesIO
+
+from cryptography.hazmat.primitives.ciphers import Cipher
+from cryptography.hazmat.primitives.ciphers.aead import AESGCM
+from cryptography.hazmat.primitives.ciphers.algorithms import AES
+from cryptography.hazmat.primitives.ciphers.modes import CBC
+from cryptography.hazmat.primitives.padding import PKCS7
+
+logger = logging.getLogger(__name__)
+
+AES_BLOCK_SIZE = 128
+ALG_CBC = "AES/CBC/PKCS5Padding"
+ALG_GCM = "AES/GCM/NoPadding"
+HEADER_ALG = "x-amz-cek-alg"
+HEADER_KEY = "x-amz-key-v2"
+HEADER_IV = "x-amz-iv"
+HEADER_MATDESC = "x-amz-matdesc"
+HEADER_TAG_LEN = "x-amz-tag-len"
+HEADER_UE_CLENGHT = "x-amz-unencrypted-content-length"
+HEADER_WRAP_ALG = "x-amz-wrap-alg"
+
+
+def is_kms_cse_encrypted(s3_metadata):
+    if HEADER_KEY in s3_metadata:
+        if s3_metadata.get(HEADER_WRAP_ALG, None) != "kms":
+            raise ValueError("Unsupported Encryption strategy")
+        if s3_metadata.get(HEADER_ALG, None) not in [ALG_CBC, ALG_GCM]:
+            raise ValueError("Unsupported Encryption algorithm")
+        return True
+    elif "x-amz-key" in s3_metadata:
+        raise ValueError("Unsupported Encryption SDK version")
```

Comment:
which SDK is this referring to?

Reply:
Ah good point, basically the SDKs used to encrypt in a way called v1 and then switched to v2: https://github.com/aws/aws-sdk-java/tree/master/aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/internal/crypto

The v2 is better as it supports CBC (aka unauthenticated encryption) and GCM (aka authenticated encryption)  while v1 was CBC only. You can recognise as one has the `x-amz-key` header and one has `x-amz-key-v2`.

In conclusion, not sure how I could name it, Encryption SDK could be confused with [this](https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/introduction.html) which is actually unrelated. 

In [this snippet](https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingClientSideEncryption.html) they call the variables "AmazonS3EncryptionClientV2".

The best way I can think of could be `Unsupported S3 Encryption Client Version`?
```suggestion
        raise ValueError("Unsupported S3 Encryption Client Version")
```